Source/Core/InputCommon/ControllerInterface/CemuHookUDPServer/CemuHookUDPServer.cpp

```diff
@@ -0,0 +1,404 @@
+
+  AddInput(new AnalogInput("Left X-", m_pad_data.left_stick_x, -128, -128));
+  AddInput(new AnalogInput("Left X+", m_pad_data.left_stick_x, 127, -128));
+  AddInput(new AnalogInput("Left Y-", m_pad_data.left_stick_y_inverted, -128, -128));
+  AddInput(new AnalogInput("Left Y+", m_pad_data.left_stick_y_inverted, 127, -128));
+  AddInput(new AnalogInput("Right X-", m_pad_data.right_stick_x, -128, -128));
+  AddInput(new AnalogInput("Right X+", m_pad_data.right_stick_x, 127, -128));
+  AddInput(new AnalogInput("Right Y-", m_pad_data.right_stick_y_inverted, -128, -128));
+  AddInput(new AnalogInput("Right Y+", m_pad_data.right_stick_y_inverted, 127, -128));
+
+  AddInput(new AnalogInput("Touch X-", m_touch_x, -TOUCH_X_AXIS_MAX));
+  AddInput(new AnalogInput("Touch X+", m_touch_x, TOUCH_X_AXIS_MAX));
+  AddInput(new AnalogInput("Touch Y-", m_touch_y, -TOUCH_Y_AXIS_MAX));
+  AddInput(new AnalogInput("Touch Y+", m_touch_y, TOUCH_Y_AXIS_MAX));
+
+  AddInput(new AnalogInput("Accelerometer Left", m_accel.x, -WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Right", m_accel.x, WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Backward", m_accel.y, -WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Forward", m_accel.y, WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Up", m_accel.z, -WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Down", m_accel.z, WiimoteEmu::ACCELEROMETER_MAX));
+
+  AddInput(new AnalogInput("Gyro Pitch Up", m_gyro.x, -WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Pitch Down", m_gyro.x, WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Roll Right", m_gyro.y, -WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Roll Left", m_gyro.y, WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Yaw Right", m_gyro.z, -WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Yaw Left", m_gyro.z, WiimoteEmu::GYROSCOPE_MAX));
+}
+
+std::string Device::GetName() const
+{
+  switch (m_model)
+  {
+  case Proto::DsModel::None:
+    return "None";
+  case Proto::DsModel::DS3:
+    return "DualShock 3";
+  case Proto::DsModel::DS4:
+    return "DualShock 4";
+  case Proto::DsModel::Generic:
+    return "Generic Gamepad";
+  default:
+    return "Device";
+  }
+}
+
+std::string Device::GetSource() const
+{
+  return "UDPServer";
+}
+
+void Device::UpdateInput()
+{
+  // Regularly tell the UDP server to feed us controller data
+  const auto now = std::chrono::steady_clock::now();
+  if (now >= m_next_reregister)
+  {
+    m_next_reregister = now + SERVER_REREGISTER_INTERVAL;
+
+    Proto::Message<Proto::MessageType::PadDataRequest> msg(m_client_uid);
+    msg.register_flags = Proto::RegisterFlags::PadID;
+    msg.pad_id_to_register = m_index;
+    msg.Finish();
+    if (m_socket.send(&msg, sizeof msg, s_server_address, s_server_port) !=
```

Comment:
Our convention is to use `sizeof(thing)`, and brace if statements when they travel over one line.